test/e2e_env/kubernetes/gateway/delegated/meshpassthrough.go

```diff
@@ -0,0 +1,143 @@
+package delegated
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	meshpassthrough_api "github.com/kumahq/kuma/pkg/plugins/policies/meshpassthrough/api/v1alpha1"
+	meshproxypatch_api "github.com/kumahq/kuma/pkg/plugins/policies/meshproxypatch/api/v1alpha1"
+	"github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func MeshPassthrough(config *Config) func() {
+	GinkgoHelper()
+
+	return func() {
+		AfterEach(func() {
+			Expect(framework.DeleteMeshResources(kubernetes.Cluster, config.Mesh, meshpassthrough_api.MeshPassthroughResourceTypeDescriptor)).To(Succeed())
+			Expect(framework.DeleteMeshResources(kubernetes.Cluster, config.Mesh, meshproxypatch_api.MeshProxyPatchResourceTypeDescriptor)).To(Succeed())
+		})
+
+		// we cannot test in a stable way first doing a successful request, because once connection is estabilished we don't break it after applying the policy (similar case to egress) the connection exists and works.
+		It("should control traffic to domains and IPs", func() {
+			esIP, err := framework.ServiceIP(kubernetes.Cluster, "external-service", config.NamespaceOutsideMesh)
+			Expect(err).ToNot(HaveOccurred())
+
+			// given
+			// we want the connection to close quickly, so after configuration change requests will start to fail.
+			meshProxyPatch := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshProxyPatch
```

Comment:
do we even need this if we first check that the request fails https://github.com/kumahq/kuma/pull/10675/files#diff-db8aa9c3fe244af0e07dc4a1c3d74aa18821196a4c3957b36ce775c6ba07d451R25 and only then allow it?

Reply:
The problem is the delay in configuration delivery. Once we apply the policy that disables traffic through passthrough, requests go through the gateway, but the configuration might not be delivered. In this case, the connection is established and will not break.

Reply:
let's leave it like this, in a perfect world we could have something like "wait until applied" (the cp would return a version of the configuration and we could check on the dp if that version was applied)